content/snowplow-recommendations/supporting-infrastructure/build.md

```diff
@@ -5,60 +5,106 @@ post = ""
+    snowflake_policy_name = "SnowflakePersonalizeIAMPolicy"
+    snowflake_role_name = "SnowflakePersonalizeIAMRole"
+    snowflake_storage_integration_name = "PersonalizeS3StorageIntegration"
+    snowflake_personalize_stage_name = "PersonalizeS3Stage"
+    ```
+    2. Set up the authentication for Terraform with your Snowflake account by providing your username and password as environment variables:
+       - `export SNOWFLAKE_USER='<username>'`
+       - `export SNOWFLAKE_PASSWORD='<password>'`
+
+    3. Run `terraform init` in this folder (i.e. `terraform_utilities/snowflake/`). This will initialise Terraform and download the necessary providers.
+
+    4. Run `terraform apply` in this directory. This will create the necessary resources in AWS and Snowflake. You will need to enter `yes` when prompted to confirm the creation of the resources.
+  
+  - Databricks (`terraform_utilities/databricks/`)
+    1. You may wish to again input the Terraform variables during the `apply` phase or provide them ahead of time by creating a `terraform.tfvars` file; the file should look like the below:
+    ```yaml
+    aws_account_id = "<AWS account ID>"
+    bucket_name = "<s3 bucket name>"
+    databricks_host = "<Databricks host url>"
+    databricks_account_id = "Databricks account ID>"
+
+    # Below variables are optional, they all have default values as listed here:
+    storage_credential_name = "storage_credential_personalize_s3"
+    storage_credential_role_name = "DatabricksStorageCredentialRole"
+    storage_credential_s3_access_policy_name = "DatabricksStorageCredentialPolicy"
+    databricks_external_catalog_name = "aws_personalize"
+    ```
+    2. Set up the authentication for Terraform with your Databricks account by exporting the environment variable with your Personal Access Token (https://docs.databricks.com/en/dev-tools/auth.html#databricks-personal-access-token-authentication):
+    - `export DATABRICKS_TOKEN='<PAT>'`
+  
+    3. Run `terraform init` in this folder (i.e. `terraform_utilities/databricks/`). This will initialise Terraform and download the necessary providers.
 
-**IV**. Run `terraform init` in the `terraform_utilities` directory. This will initialise terraform and download the necessary providers.
+    4. Due to the nature of the required role on AWS for the Databricks Storage Credentials be self referential and the current limiting factor of Terraform to create a trust policy on a new role in the same `apply` phase, the TF file for Databricks will need to be applied twice. So please run `terraform apply` and you'll see the error regarding the failure in creating the external location because the AWS IAM role does not have the READ permission on the S3 bucket TF would have just created. Upon running `terraform apply` a second time this will be amended and the external location as well as the new external catalog (with a managed location) will both be successfully created.
```

Comment:
Nit- this sentence is a bit unclear:
Due to the nature of the required role on AWS for the Databricks Storage Credentials being self referential and the current limitation of Terraform to not be able to create a trust policy on a new role in the same `apply` phase, the TF file for Databricks will need to be applied twice.

maybe something like:
Because the AWS role required for the Databricks Storage Credentials is self-referential and Terraform currently cannot create a trust policy for a new role in the same apply phase, the TF file for Databricks needs to be applied twice.